clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -952,6 +954,32 @@ public String toString() {
         }
     }
 
+    public static class LambdaValidator implements Validator {
+        BiConsumer<String, Object> ensureValid;
+        Supplier<String> toStringFunction;
+
+        private LambdaValidator(BiConsumer<String, Object> ensureValid,
+                                Supplier<String> toStringFunction) {
+            this.ensureValid = ensureValid;
+            this.toStringFunction = toStringFunction;
+        }
+
+        public static LambdaValidator with(BiConsumer<String, Object> ensureValid,
```

Comment:
Instead of BiConsumer here, and relying on exceptions to pass validation errors back to the caller, maybe this should be `(k, v) -> error`?

Reply:
The current form maps exactly to `ensureValid` in the Validator interface. This addition here is minor. I'd suggest discussion any improvements on config defs in a separate PR/jira issue.